### ISSUE TITLE
Null reference exception in Contains(otherEnumerable)

### ISSUE THREAD

**Issue by NickDarvey** [tpierrain/NFluent#251]
[NFluent.EnumerableCheckExtensions.Contains[TU](ICheck`1 check, IEnumerable otherEnumerable)](https://github.com/tpierrain/NFluent/blob/c62be6fd4d2fd7f24b5dacf4260f2b806a46bb2d/src/NFluent/Checks/EnumerableCheckExtensions.cs#L63) throws a null reference exception for a failing test case.

## Expected Behavior
```
Message: NFluent.FluentCheckException : 
The checked enumerable does not contain the expected value(s):
	["c", "d"]
The checked enumerable:
	["a", "b"]
The expected value(s):
	["c", "d"]
```

## Current Behavior
```
System.NullReferenceException : Object reference not set to an instance of an object.
```

## Code that reproduce the issue
```csharp
var x = new[] { "a", "b" }.AsEnumerable();
var y = new[] { "c", "d" }.AsEnumerable();

Check.That(x).Contains(y);
```

## Context
This happens in 2.3.0, but not 2.2.0.
A passing test case works fine.
```
at NFluent.Helpers.ExceptionHelper.ExceptionScanner(String assemblyMarker, String nameSpace, String assertionExceptionName, String ignoreExceptionName, String inconclusiveExceptionName)
   at NFluent.Helpers.ExceptionHelper.get_Constructors()
   at NFluent.Helpers.ExceptionHelper.BuildException(String theMessage)
   at NFluent.Kernel.ExceptionReporter.ReportError(String message)
   at NFluent.EnumerableCheckExtensions.Contains[TU](ICheck`1 check, IEnumerable otherEnumerable)
```

## Your Environment
* Version used: 2.3.0
* Framework version: netcoreapp2.0

**Comment by dupdob** [comment]
Hi
Sorry that you have his issue. Can you try with another failing check? It looks like something related to the te framework discovery mechanism, so it probably happens for each error.
Are you using xUnit? nUnit ? mstest?

**Comment by NickDarvey** [comment]
I disabled 'Just My Code' and I'm seeing this:
`System.ArgumentException: 'FrameworkName cannot have less than two components or more than three components.'` from `System.Runtime.Versioning.FrameworkName`.

Here's my project file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
    
    <IsPackable>false</IsPackable>

    <AssemblyName>NickDarvey.ServiceFabric.EventHubs.AspNetCore.IntegrationTests</AssemblyName>

    <RootNamespace>ServiceFabric.EventHubs</RootNamespace>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.2" />
    <PackageReference Include="Moq" Version="4.8.2" />
    <PackageReference Include="NFluent" Version="2.3.0" />
    <PackageReference Include="ServiceFabric.Mocks" Version="3.2.0" />
    <PackageReference Include="xunit" Version="2.3.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.3" />
    <PackageReference Include="Microsoft.AspNetCore.Http" Version="2.0.3" />
    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\ServiceFabric.EventHubs.AspNetCore\ServiceFabric.EventHubs.AspNetCore.csproj" />
  </ItemGroup>

</Project>
```

**Comment by dupdob** [comment]
Thanks for the detailed report. I have created a netcore 2.0 project using xUnit and I fail to reproduce.
I am not giving up, so don't worry.
Thanks for reporting the System.Argument exception, but this is not related to NFluent, [see here](https://developercommunity.visualstudio.com/content/problem/224359/systemargumentexception-frameworkname-cannot-have.html). Could you please try and move forward and see what happens next?

**Comment by dupdob** [comment]
Thanks for the detailed report. I have created a netcore 2.0 project using xUnit and I fail to reproduce.
I am not giving up, so don't worry.
Thanks for reporting the System.Argument exception, but this is not related to NFluent, [see here](https://developercommunity.visualstudio.com/content/problem/224359/systemargumentexception-frameworkname-cannot-have.html). Could you please try and move forward and see what happens next?

**Comment by dupdob** [comment]
I fear that me trying to reproduce your issue may turn to a wild goose chase for me, so I will publish a package with debug information so I can pinpoint which line is raising this issue.

**Comment by dupdob** [comment]
Hello
@NickDarvey : I have published a package with pdb files here
https://www.myget.org/feed/dupdobnightly/package/nuget/NFluent

you can install it like this:
```
PM> Install-Package NFluent -Version 2.3.1-beta-0173 -Source https://www.myget.org/F/dupdobnightly/api/v3/index.json 
```

Can you please try ? it should give you the line number causing the issue.

**Comment by dupdob** [comment]
Hello again
We have been able to reproduce this issue and I have published a beta with the fix.
It would be great if you try it (it’s 2.3.1-beta176 on myGet) and confirm it is fixed on your side as well.
We will publish a fix soon

**Comment by NatMarchand** [comment]
I had the error on 2.3.0, seems fixed with the 2.3.1-beta176
Thanks !

**Comment by NickDarvey** [comment]
@dupdob that did the trick, thank you!

(I'll leave it up to you to close the issue or leave it open till 2.3.1 is published.)

**Comment by dupdob** [comment]
Fixed in 2.3.1. Thanks everyone
